hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/lock/granular/OmLockManager.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.ozone.om.lock.granular;
+
+import com.google.common.util.concurrent.StripedLock;
+import com.google.common.util.concurrent.StripedReadWriteLocks;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.apache.hadoop.ozone.om.lock.granular.OmComponentLock.Component;
+import org.apache.hadoop.ozone.om.lock.granular.OmComponentLock.Type;
+import org.apache.ratis.util.UncheckedAutoCloseable;
+
+/**
+ * Manage locking of volume, bucket and keys.
+ */
+public class OmLockManager {
+  private final StripedReadWriteLocks volumeLocks =  StripedReadWriteLocks.newInstance(1 << 10, true);
+  private final StripedReadWriteLocks bucketLocks = StripedReadWriteLocks.newInstance(1 << 12, true);
+  private final StripedReadWriteLocks keyLocks = StripedReadWriteLocks.newInstance(1 << 16, true);
+
+  static OmComponentLock newOmComponentLock(Component component, Type type, StripedLock<String> lock) {
+    return new OmComponentLock(lock.getStripeKey(), component, type, lock.getIndex(), lock.getLock());
+  }
+
+  private OmComponentLock newVolumeLock(Type type, String name) {
+    return newOmComponentLock(Component.VOLUME, type, volumeLocks.get(name));
```

Comment:
This is a strange implementation because this class already controls the locks. Now it is passing the same lock to multiple of these new lock objects. It would be more straightforward for this class to a handle the mapping of components to locks without splitting this responsibility across two classes which are sharing lock instances.

Reply:
Sorry, I don't get this comment.  Questions below:

> ... Now it is passing the same lock to multiple of these new lock objects.  ...

Where are the  "multiple of these new lock objects"

> ...  splitting this responsibility across two classes ...

What is the responsibility and which two classes?